diff --git a/vendor/thrift/CMakeLists.txt b/vendor/thrift/CMakeLists.txt
index caf98fd..c7b71e2 100644
--- a/vendor/thrift/CMakeLists.txt
+++ b/vendor/thrift/CMakeLists.txt
@@ -4,6 +4,7 @@ set(LIBRARY_DIR "${CMAKE_CURRENT_SOURCE_DIR}/lib/cpp")
 set(thriftcpp_SOURCES
     "${LIBRARY_DIR}/src/thrift/TApplicationException.cpp"
     "${LIBRARY_DIR}/src/thrift/TOutput.cpp"
+    "${LIBRARY_DIR}/src/thrift/TUuid.cpp"
     "${LIBRARY_DIR}/src/thrift/async/TAsyncChannel.cpp"
     "${LIBRARY_DIR}/src/thrift/async/TAsyncProtocolProcessor.cpp"
     "${LIBRARY_DIR}/src/thrift/async/TConcurrentClientSyncInfo.h"
